### PR TITLE
Allow specific buid result selection of whether to archive artifacts

### PR DESCRIPTION
**Pull request by ScoobyChris**

### CODE REVIEW

**Comment by jtnord** [comment]
What If I also want to archive on unstable results?

I'm not sure this is the correct way to do this.  Seems like it is more a job for the [Flexible Publish plugin](https://wiki.jenkins-ci.org/display/JENKINS/Flexible+Publish+Plugin)?

**Comment by oleg-nenashev** [comment]
There was a discussion in #1912 
Personally I'm fine with having a bit more functionality in the default publisher

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,4 @@ jenkins_*.changes
 push-build.sh
 war/node_modules/
 .java-version
+/bin/
```
Comment (daniel-beck) [inline .gitignore]: What is this change?

**Comment by daniel-beck** [comment]
*Unstable/Stable* are not distinguished. Seems more important than being able to archive on build abort IMO.

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java [hunk 2]
```diff
@@ -169,8 +172,34 @@ public boolean isOnlyIfSuccessful() {
...
+    }
+
+    public boolean isOnAborted() {
+        return onAborted;
+    }
+
     @DataBoundSetter public final void setOnlyIfSuccessful(boolean onlyIfSuccessful) {
-        this.onlyIfSuccessful = onlyIfSuccessful;
+        this.onSuccessful = onlyIfSuccessful;
+        this.onFailed = false;
```
Comment (daniel-beck) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: This actually changes the behavior. When it was *disabled*, it archived *always*.

Why isn't this `@Deprecated` to begin with?

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java [hunk 3]
```diff
@@ -89,6 +89,9 @@
      * Archive only if build is successful, skip archiving on failed builds.
      */
     private boolean onlyIfSuccessful;
```
Comment (daniel-beck) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: Should probably be `transient Boolean` so it doesn't get persisted again, and there should be `readResolve` that transforms the loaded value into corresponding `onSuccessful`/`onFailed`/`onAborted` values. This apparently has no support for loading existing configurations (and behaving correctly) at all.

**Comment by daniel-beck** [comment]
:-1: Appears to break backwards compatibility.

**Comment by daniel-beck** [comment]
PR appears to have been abandoned two years ago, closing.

**Pull request closed without merging**